ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -745,7 +746,9 @@ module Winbind = struct
   let random_string len =
     let upper_char_start = 65 in
```

Comment:
`Char.code('A')` would be more readable

Reply:
Do we have to make sure this is random enough? The range of random characters is quite limited and if the string is also short, it could be enumerated. How important is this from a security perspective?

Reply:
Every join domain will create a machine account and coresponding passwrod in domain controller, 
The mache account name lengh is limitted to 15 chars.

If the  hostname longger than 15 bits, we will construct one
`7_chars_of_prefix_of_hostname`-`7_chars_of_random`

Yes, the random characters is limitted,  (26*26*26*26*26*26*26),  
but we  only need the it when the host name longger than 15, and the how many depends on how many users join domain. I do not expect customer can join domain so many times.
BTW, PBIS do the same way.
This does not related to security.